tests/crm/test_crm.py

```diff
@@ -0,0 +1,914 @@
+import pytest
+import time
+import re
+import json
+import ipaddress
+import netaddr
+import copy
+
+from jinja2 import Template
+from tests.common.plugins.loganalyzer.loganalyzer import LogAnalyzer
+from tests.common.helpers.assertions import pytest_assert
+from collections import OrderedDict
+
+
+logger = logging.getLogger(__name__)
+
+CRM_POLLING_INTERVAL = 1
+CRM_UPDATE_TIME = 4
+SONIC_RES_UPDATE_TIME = 50
+
+THR_VERIFY_CMDS = OrderedDict([
+    ("exceeded_used", "bash -c \"crm config thresholds {{crm_cli_res}}  type used; crm config thresholds {{crm_cli_res}} low {{crm_used|int - 1}}; crm config thresholds {{crm_cli_res}} high {{crm_used|int}}\""),
+    ("clear_used", "bash -c \"crm config thresholds {{crm_cli_res}} type used && crm config thresholds {{crm_cli_res}} low {{crm_used|int}} && crm config thresholds {{crm_cli_res}} high {{crm_used|int + 1}}\""),
+    ("exceeded_free", "bash -c \"crm config thresholds {{crm_cli_res}} type free && crm config thresholds {{crm_cli_res}} low {{crm_avail|int - 1}} && crm config thresholds {{crm_cli_res}} high {{crm_avail|int}}\""),
+    ("clear_free", "bash -c \"crm config thresholds {{crm_cli_res}} type free && crm config thresholds {{crm_cli_res}} low {{crm_avail|int}} && crm config thresholds {{crm_cli_res}} high {{crm_avail|int + 1}}\""),
+    ("exceeded_percentage", "bash -c \"crm config thresholds {{crm_cli_res}} type percentage && crm config thresholds {{crm_cli_res}} low {{th_lo|int}} && crm config thresholds {{crm_cli_res}} high {{th_hi|int}}\""),
+    ("clear_percentage", "bash -c \"crm config thresholds {{crm_cli_res}} type percentage && crm config thresholds {{crm_cli_res}} low {{th_lo|int}} && crm config thresholds {{crm_cli_res}} high {{th_hi|int}}\"")
+])
+
+EXPECT_EXCEEDED = ".* THRESHOLD_EXCEEDED .*"
+EXPECT_CLEAR = ".* THRESHOLD_CLEAR .*"
+
+RESTORE_CMDS = {"test_crm_route": [],
+                "test_crm_nexthop": [],
+                "test_crm_neighbor": [],
+                "test_crm_nexthop_group": [],
+                "test_acl_entry": [],
+                "test_acl_counter": [],
+                "test_crm_fdb_entry": [],
+                "crm_cli_res": None,
+                "wait": 0}
+
+
+@pytest.fixture(scope="module", autouse=True)
+def crm_thresholds(duthost):
```

Comment:
Is it any reason why we store fixtures in test suite file?
My suggestion:
if we have local `conftest.py` we should there all fixtures and in test suite file we should store test functions and related helpers functions

Reply:
In exited test suites there are many files which has defined fixtures inside test suite files.
However I agree with your suggestion, now its fixed.